security_privacy_ln.asciidoc

```diff
@@ -0,0 +1,614 @@
+A Lightning node only knows the balances of its adjacent channels.
+The protocol provides no standard way to query the balance of a remote channel.
+
+However, an attacker can reveal the balance of a remote channel in a _probing attack_.
+In information security, probing refers to the technique of sending requests to a targeted system and making conclusions about its private state based on the received responses.
+
+Lightning channels are prone to probing.
+Recall that a standard Lightning payment starts with the receiver creating a random payment secret and sending its hash to the sender.
+Note for the intermediary nodes, all hashes look random.
+There is no way to tell whether a hash corresponds to a real secret or was generated randomly.
+
+The probing attack proceeds as follows.
+Say, the attacker Mallory wants to reveal Alice's balance of a public channel between Alice and Bob.
+Suppose the total capacity of that channel is 1 million satoshis.
+Alice's balance could be anything from zero to 1 million satoshis (to be precise, the estimate is a bit tighter due to channel reserve, but we don't account for it here for simplicity).
+Mallory opens a channel with Alice with 1 million satoshis and sends 500 thousand satoshis to Bob via Alice using a _random number_ as the payment hash.
+Of course, this number does not correspond to any known payment secret.
+Therefore, the payment will fail.
+The question is: how exactly will it fail?
+
+There are two scenarios.
+If Alice has more than 500 thousand satoshis on her side of the channel to Bob, she forwards the payment.
+Bob decrypts the payment onion and realizes that the payment is intended for him.
+He looks up his local store of payment secrets, searches for the preimage that corresponds to the payment hash but does not find one.
+Following the protocol, Bob returns the "unknown payment hash" error to Alice, who relays it back to Mallory.
+As a result, Mallory knows that the payment _could have succeeded_ if the payment hash was real.
+Therefore, Mallory can update her estimation of Alice's balance from "between zero and 1 million" to "between 500 thousand and one million."
+Another scenario happens if Alice's balance is lower than 500 thousand satoshis.
+In that case, Alice is unable to forward the payment and returns the "insufficient balance" error to Mallory.
+Mallory updates her estimation from "between zero and 1 million" to "between zero and 500 thousand."
+
+Note that in any case, Mallory's estimation becomes twice as precise after just one probing!
+She can continue probing, choosing the next probing amount such that it divides the current estimation interval in half.
+This well-known search technique is called _binary search_.
+With binary search, the number of probes is _logarithmic_ in the desired precision.
+For example, to obtain Alice's balance in a channel of 1 million satoshis up to a single satoshi, Mallory would only have to perform stem:[\log_2(1000000) \approx 20] probings.
```

Comment:
This formula doesn't get previewed here: https://github.com/seresistvanandras/lnbook/blob/lnsecurityprivacy/security_privacy_ln.asciidoc#linking-senders-and-receivers

Reply:
Indeed, you're right. Locally, in my text editor, it renders correctly. Maybe, it's a fault of GitHub? Anyone could confirm this? cc @s-tikhomirov  @carlaKC

Reply:
This is a Github rendering limitation, but our asciidoc rendered will handle it correctly